docs/toolbox.md

```diff
@@ -0,0 +1,213 @@
+
+## Full and Compact Modes
+
+Toolbox supports full mode (illustrated above) and compact mode. In compact mode, element names are hidden. To see an individual element name, a user should move the mouse pointer over the element icon.
+
+<img src="./images/survey-creator-toolbox-compact.png" alt="Survey Creator - Toolbox in compact mode" width="50%">
+
+Toolbox switches between the modes automatically based on available width. Specify the [`forceCompact`](https://surveyjs.io/Documentation/Survey-Creator?id=questiontoolbox#forceCompact) property if you want the toolbox to always use a specific mode:
+
+```js
+// Compact mode
+creator.toolbox.forceCompact = true;
+// Full mode
+creator.toolbox.forceCompact = false;
+```
+
+You can also use the [`isCompact`](https://surveyjs.io/Documentation/Survey-Creator?id=questiontoolbox#isCompact) property to find out whether the toolbox is currently in compact mode:
+
+```js
+console.log(creator.toolbox.isCompact);
+```
+
+## Limit Available Question and Panel Types
+
+All available question and panel types are listed in the [`getType()`](https://surveyjs.io/Documentation/Library?id=Question#getType) method description. If you need to show only a part of these types, specify them in the Survey Creator's [`questionTypes`](https://surveyjs.io/Documentation/Survey-Creator?id=surveycreator#questionTypes) array:
+
+```js
+const creatorOptions = {
+    questionTypes: ["text", "checkbox", "radiogroup", "dropdown"]
+};
+
+const creator = new SurveyCreator.SurveyCreator(creatorOptions);
+
+// In modular applications
+import { SurveyCreator } from "survey-creator-knockout";
+// or
+import { SurveyCreator } from "survey-creator-react";
+const creator = new SurveyCreator(creatorOptions);
+```
+
+[View Toolbox Customization example](https://surveyjs.io/Examples/Survey-Creator?id=toolboxcustomization)
+
+## Group Toolbox Items by Categories
+
+> NOTE: Compact toolbox does not display categories.
+
+To group toolbox items, call the [`changeCategories()`](https://surveyjs.io/Documentation/Survey-Creator?id=questiontoolbox#changeCategories) method. It accepts an array of objects with the following fields:
+
+- `name`        
+The name of the item that should be grouped. Refer to the [`getType()`](https://surveyjs.io/Documentation/Library?id=Question#getType) method description for a list of accepted values.
+
+- `category`      
+A category for this item.
+
+The following code places the [Panel](https://surveyjs.io/Documentation/Library?id=panelmodel) and [Panel Dynamic](https://surveyjs.io/Documentation/Library?id=questionpaneldynamicmodel) types into the Panels category and the [Matrix](https://surveyjs.io/Documentation/Library?id=questionmatrixmodel), [Matrix Dropdown](https://surveyjs.io/Documentation/Library?id=questionmatrixdropdownmodel), and [Matrix Dynamic](https://surveyjs.io/Documentation/Library?id=questionmatrixdynamicmodel) types into the Matrices category:
```

Comment:
Not a big deal... "Matrixes" and "Indexes" are more commonly used than "Matrices" and "Indices". 

```suggestion
The following code places the [Panel](https://surveyjs.io/Documentation/Library?id=panelmodel) and [Panel Dynamic](https://surveyjs.io/Documentation/Library?id=questionpaneldynamicmodel) types into the Panels category and the [Matrix](https://surveyjs.io/Documentation/Library?id=questionmatrixmodel), [Matrix Dropdown](https://surveyjs.io/Documentation/Library?id=questionmatrixdropdownmodel), and [Matrix Dynamic](https://surveyjs.io/Documentation/Library?id=questionmatrixdynamicmodel) types into the Matrixes category:
```